src/explore-education-statistics-frontend/src/modules/table-tool/components/TimePeriodDataTable.tsx

```diff
@@ -24,156 +24,167 @@ interface Props {
   locations: LocationFilter[];
   results: TableData['result'];
   footnotes?: TableData['footnotes'];
+  tableHeadersConfig?: TableHeadersFormValues;
 }
 
-const TimePeriodDataTable = (props: Props) => {
-  const {
-    filters,
-    timePeriods,
-    locations,
-    indicators,
-    results,
-    footnotes,
-  } = props;
-
-  const dataTableRef = useRef<HTMLTableElement>(null);
-
-  const [tableHeaders, setTableHeaders] = useState<TableHeadersFormValues>({
-    columnGroups: [],
-    columns: [],
-    rowGroups: [],
-    rows: [],
-  });
-
-  const removeSiblinglessTotalRows = (
-    categoryFilters: Dictionary<CategoryFilter[]>,
-  ): CategoryFilter[][] => {
-    return Object.values(categoryFilters).filter(filter => {
-      return filter.length > 1 || !filter[0].isTotal;
+const TimePeriodDataTable = forwardRef<HTMLElement, Props>(
+  (props: Props, dataTableRef) => {
+    const {
+      filters,
+      timePeriods,
+      locations,
+      indicators,
+      results,
+      footnotes,
+      tableHeadersConfig,
+    } = props;
+
+    const [tableHeaders, setTableHeaders] = useState<TableHeadersFormValues>({
```

Comment:
As we're getting all of the table header configuration as a prop (this should be the single source of truth), we shouldn't be setting it as state as well. 

This is a fairly common anti-pattern in React application as there is duplication of state which means we have to write more code to keep the props and state in sync (and is potentially buggy). We should avoid this whenever possible.

Reply:
[As discussed above](https://github.com/dfe-analytical-services/explore-education-statistics/pull/559#discussion_r313323040), This is to allow the component to either render with a given configuration or fallback to a default ~ in the table tool example, it's provided a configuration that *is default*, however the table tool page also has the ability to alter the provided configuration with the TableHeadersForm component.

Perhaps we should discuss further